src/snap/grpc_export.cc

```diff
@@ -93,6 +129,30 @@ future<void> Plugin::GRPCExportImpl::DoExport(shared_ptr<PluginInterface> plugin
     return std::async(std::launch::deferred, [=](){ self->doJoin(); });
 }
 
+std::shared_ptr<grpc::ServerCredentials> Plugin::GRPCExportImpl::configureCredentials() {
+  auto retVal = grpc::InsecureServerCredentials();
+  if (meta->tls_cipher_suites != "") {
+    std::cout << "Overriding cipher suites with '" << meta->tls_cipher_suites << "'" << std::endl;
+    setenv("GRPC_SSL_CIPHER_SUITES", meta->tls_cipher_suites.c_str(), 1);
+  }
+  if (meta->tls_enabled) {
+    auto tls_key = load_key(this->meta->tls_certificate_key_path);
+    auto tls_crt = load_key(this->meta->tls_certificate_crt_path);
+    auto tls_ca = load_key(this->meta->tls_certificate_authority_path);
+
+    grpc::SslServerCredentialsOptions::PemKeyCertPair pkcp;
+    pkcp = {tls_key, tls_crt};
+
+    grpc::SslServerCredentialsOptions ssl_opts;
+    ssl_opts.pem_key_cert_pairs.push_back(pkcp);
+    ssl_opts.pem_root_certs = tls_ca;
```

Comment:
TLS support has to allow multiple root certificates. Plugin (library) argument must support a list of locations. see:
* Snap documentation: [SECURE_PLUGIN_COMMUNICATION.md#using-system-installed-ca-certificates](https://github.com/intelsdi-x/snap/blob/master/docs/SECURE_PLUGIN_COMMUNICATION.md#using-system-installed-ca-certificates)
* [GRPC documentation](https://grpc.io/grpc/cpp/structgrpc_1_1_ssl_credentials_options.html)
* plugin-lib-go [implementation in readRootCAs()](https://github.com/intelsdi-x/snap-plugin-lib-go/blob/cdca5a915f3fb0f34095d2c6d85eebda4313c24f/v1/plugin/plugin.go#L216), [plugin flags documentation - see root-cert-paths](https://github.com/intelsdi-x/snap-plugin-lib-go/blob/fa976596b024ed9d0711a3514436f14fc1ecf63c/README.md#plugin-flags).